9 спринт - PlaylistMaker

who7cares opened the pull request:
9 спринт

Commit:
- изменена тема в AndroidManifest.xml
- добавлены шрифты, стили

Commit:
-добавлена темная тема (без глюков)

Commit:
добавлен TextInputEditText в SearchActivity

Commit:
реализованы интенты в SettingsActivity (кроме свича)

Commit:
сохранение поисковвой строки в SearchActivity.kt после поворота экрана

(для чего тут методы onSaveInstanceState и onRestoreInstanceState?)

IliaAnK commented:
Рекомендация:
Папку `.idea` стоит добавить в `gitignore` файл. И удалить её из гитхаб репозитория

IliaAnK commented on app/src/main/res/values/style.xml:
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="buttonStyle">
```
Обязательно:
Стиль кнопок для «Главного экрана» должен наследоваться от `Widget.MaterialComponents.Button.UnelevatedButton`

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -2,12 +2,10 @@
-    >
+    android:padding="@dimen/screen_margins" >
 
     <LinearLayout
```
В качестве заголовка экрана лучше использовать `Toolbar`. Это стандартный компонент. В него можно передать текст заголовка, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.
Для остальных экранов **аналогично**.

IliaAnK commented on app/src/main/java/com/bignerdranch/playlistmaker/SearchActivity.kt:
```diff
@@ -1,16 +1,140 @@
+    @SuppressLint("ClickableViewAccessibility")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enableEdgeToEdge()
```
Обязательно:
Нужно либо отключить EdgeToEdge режим, либо адаптировать под него приложение.
Сейчас зона `Toolbar` заезжает под `AppBar` из-за чего экран настроек перестаёт соответствовать макетам и невозможно нажать кнопку "Назад"
Аналогично для остальных экранов

IliaAnK commented on app/src/main/java/com/bignerdranch/playlistmaker/SearchActivity.kt:
```diff
@@ -1,16 +1,140 @@
+        return super.dispatchTouchEvent(event)
     }
+
+    val simpleTextWatcher = object : TextWatcher {
```
Рекомендация:
`private val`

IliaAnK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -4,16 +4,64 @@
+            android:textSize="@dimen/text" />
+    </LinearLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
```
Обязательно:
Согласно требованиям спринта нужно использовать реализацию с `FrameLayout` внутри которого должен находится `EditText` и `ImageView`. Обращу внимание, что `ImageView` должен находится поверх поля ввода.

who7cares replied:
'На экране «Поиск» в поле ввода иконка «Очистить» и её размер не соответствует макетам.'
в фигме ни на одном из макетов не нашел кнопки очистить

IliaAnK replied:
Ссылка на макеты из задания:
[Светлая](https://www.figma.com/design/77fvrLWpv9rXEhBoubXczZ/Playlist-Maker.-Search.-Light-Mode-(YP)?node-id=1-370&t=3Xls3rgAKIxOokws-4) и [тёмная](https://www.figma.com/design/sRkCvWJ7uYaeINvSed3lA6/Playlist-Maker-%2F-Search.-Dark-Mode?node-id=1-370&t=IpPoyZalJS3USZv6-4) темы

<img width="363" alt="Снимок экрана 2024-12-10 в 7 24 20 PM" src="https://github.com/user-attachments/assets/a4962509-75d1-4822-a717-c2b57794b94e">

who7cares replied:
спасибо

IliaAnK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -4,16 +4,64 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@color/white"
+    android:background="@color/search_backgroudnd"
```
Рекомендация:
search_backgrou**nd**

IliaAnK commented on app/src/main/java/com/bignerdranch/playlistmaker/SearchActivity.kt:
```diff
@@ -1,16 +1,140 @@
+    }
+
+//    // Сохраняем данные в Bundle при изменении конфигурации (например, при повороте экрана)
+//    override fun onSaveInstanceState(outState: Bundle) {
```
Обязательно:
Согласно требованиям спринта реализация метода `onSaveInstanceState` и `onRestoreInstanceState ` обязательна.

При условии, что выше, в методе `onCreate ` есть восстановление состояние, то реализацию метода `onSaveInstanceState` можем вынести в рекомендации

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -40,12 +40,8 @@
         android:paddingBottom="34dp">
```
Рекомендация к каждой конструкции
```
<LinearLayout> 
    <TextView/>
    <Switch или ImageView/> 
</LinearLayout>
```
Во-первых, это излишняя вложенность, которая негативно сказывается при отрисовке фрейма.
Во-вторых, она вызывает трудности с понимание того как свёрстан экран;
В-третьих, проще формировать и применять стили.

Каждая такая конструкция может быть представлена одной View.
Для свичера:
```
<com.google.android.material.switchmaterial.SwitchMaterial
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:gravity="center_vertical"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/dark_theme"
    android:textSize="16sp" />
```
Для TextView с иконкой справа:
```
<com.google.android.material.textview.MaterialTextView
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:drawableEnd="@drawable/share"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/share_with_app"
    android:textSize="16sp" />
```

Общие характеристика вынести само собой можно вынести в стили.

Итого иерархия всего `layout`-а будет вот такой:
```
<LinearLayout>
    <MaterialToolbar/>
    <SwitchMaterial/>
    <MaterialTextView/>
    <MaterialTextView/>
    <MaterialTextView/>
<LinearLayout/>
```

Commit:
я пол недели настраивал TextInputLayout, чтобы он работал как надо...дайте пожить спокойно....

IliaAnK commented:
> я пол недели настраивал TextInputLayout, чтобы он работал как надо...

Я у верен, что знания которые ты получил во время этой работы не пройдут напрасно, но согласно заданию нужно поработать с `FrameLayout`, `EditText` и `ImageView`.

Commit:
правки:
(TextInputLayout заменен на FrameLayout и EditText)
(исправлен onSaveInstanceState и onRestoreInstanceState)
(мелкие правки по макету)

IliaAnK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -32,40 +32,42 @@
             android:inputType="text"
+            android:maxLines="1"
+
+            android:paddingLeft="12dp"
```
paddingStart

IliaAnK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -32,40 +32,42 @@
+
+            android:paddingLeft="12dp"
+            android:paddingTop="8dp"
+            android:paddingRight="40dp"
```
paddingEnd

Commit:
правки:
(логика отображения и очистки closeImageView в SearchActivity)

IliaAnK commented on app/src/main/java/com/bignerdranch/playlistmaker/SearchActivity.kt:
```diff
@@ -92,7 +95,9 @@ class SearchActivity : AppCompatActivity() {
         }
 
         override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            closeImageView.visibility = if (p0.isNullOrEmpty()) View.GONE else View.VISIBLE
```
Рекомендация:
`closeImageView.isVisible =  !p0.isNullOrEmpty()`
